R/utilities-molecule-plots.R

```diff
@@ -315,3 +337,34 @@ getDualAxisPlot <- function(leftPlotObject, rightPlotObject) {
 
   return(mergedPlotObject)
 }
+
+#' @title .addLLOQLayer
+#' @description Add a line layer representing the Lower Limit Of Quantification (LLOQ)
+#' @param plotObject A `ggplot` object
+#' @param data A data.frame with labels mapped to properties and obtained from a `DataMapping` object
+#' @param mapLabels List of mapped label names passed to `ggplot2::aes_string`
+#' @return A `ggplot` object
+#' @keywords internal
+.addLLOQLayer <- function(plotObject, data, mapLabels) {
+  aestheticValues <- .getAestheticValuesFromConfiguration(
+    n = 1,
+    position = 0,
+    plotConfigurationProperty = plotObject$plotConfiguration$lines,
+    propertyNames = c("color", "linetype", "size", "alpha")
+  )
+
+  plotObject <- plotObject +
+    ggplot2::geom_hline(
+      data = data,
+      mapping = ggplot2::aes(
+        yintercept = .data[[mapLabels$lloq]],
+        color = .data[[mapLabels$color]]
+      ),
+      linetype = "dotted",
```

Comment:
Might be better to make it a `tlfEnv` parameter (in `tlf-env.R`) in case users plan to change it to something else.

Reply:
@pchelle Shouldn't it be included in `PlotConfiguration` ?

Reply:
PlotConfiguration `lines` properties are already used for the simulated time profile.
But you can indeed define a similar field named `lloq` in the plot configuration and make the lloq lines get these properties using something like this in the PlotConfiguration:
```r
private$.lloq <- ThemeAestheticSelections$new(linetype = Linetypes$dotted)
```
And then just use the lloq property in `.addLLOQLine`
```r
aestheticValues <- .getAestheticValuesFromConfiguration(
    n = 1,
    position = 0,
    plotConfigurationProperty = plotObject$plotConfiguration$lloq,
    propertyNames = c("color", "linetype", "size", "alpha")
  )
```
Making `aestheticValues$linetype` "dotted" by default

Reply:
I added it tlfEnv in 819ef5b